### DISCUSSION TITLE
rustic 0.6.0

### DISCUSSION THREAD

**Discussion by aawsome** [rustic-rs/rustic#916] [Announcements]
Finally, rustic 0.6.0 is released.

This is the first release using the rustic_core crate for basically all rustic functionality. This was a very big refactor, many thanks go to @simonsan!
Even though we took great care to keep all functionality, please keep a good eye on this release due to the big changes!

### Breaking Changes

- We refactored to
  [`rustic_core`](https://www.github.com/rustic-rs/rustic_core). This means that
  most of the underlying logic can now be used as a library. The CLI is now a
  thin wrapper around the library. This also means that the CLI is now much more
  customizable. Please check out the
  [documentation](https://rustic.cli.rs/docs/getting_started.html) for more
  information.

### Bug Fixes

- Retrying backend access didn't work for long operations. This has been fixed
  (and retries are now customizable)
- Prune did abort when no time was set for a pack-do-delete. This case is now
  handled correctly.
- The zstd compression library led to data corruption in very unlikely cases.
  This has been fixed by a dependency update.
- The glob option did only work with absolute files. This has been fixed.
- Non-unicode link targets are now correctly handled on Unix (after this has
  been added to the restic repo format).
- The `--dry-run` option now works as expected in the `init` command.

### Features

- New global configuration paths are available, located at `/etc/rustic/*.toml`
  or `%PROGRAMDATA%/rustic/config/*.toml`, depending on your platform.
- REST backend: Now allows to use custom TLS root certificates.
- Environment variables for programms called by rustic can now be set in the
  config files.
- Creation of new keys now enforces confirmation of entered key. This helps to
  prevent mistype of passwords during the initial entry
- Wait for password-command to successfully exit, allowing to input something
  into the command, and read password from stdout.
- backup: New option --init to initialize repository if it doesn't exist yet.
- backup: New option `no-require-git` - if enabled, a git repository is not
  required to apply `git-ignore` rule.
- restore: The restore algorithm has been improved and should now be faster for
  remote repositories.
- restore: Files are now allocated just before being first processed. This
  allows easier resumed restores.
- repoinfo: Added new options --json, --only-files, --only-index.
- check: Add check if time is set for packs-to-delete.
- ls: Options --long (-l) and --summary (-s) have been added.
- forget: Option --json has been added.

<hr /><em>This discussion was created from the release <a href='https://github.com/rustic-rs/rustic/releases/tag/v0.6.0'>rustic 0.6.0</a>.</em>

**Comment by jiripospisil** [comment]
Hello, will pre-built binaries be available?

**Reply by aawsome** [reply]
Yes, soon - we changed quite some things in the CI pipelines, automatic binary builds didn't work as I expected. But we'll fix this - and of course publish pre-build binaries...

**Reply by rehanone** [reply]
Any update on the binaries?

**Reply by simonsan** [reply]
@jiripospisil @rehanone We are on it, keep in mind that we are working in our spare time on this project and have families and children. :)

In the meantime you can download nightly binaries from here, the latest one is 4 hours old:
https://github.com/rustic-rs/nightly

Cheers!

**Reply by rehanone** [reply]
@aawsome , fully understand and appreciate the work of all the contributes here. While this release issue is being worked on, is it possible to mark this release as draft? The reason is that it is breaking the automation to install the binaries from github which takes the latest release tag and tries to pull the binaries which are not found.

**Reply by usmannasir** [reply]
Yes guys really appreciate the work you all do, but if you guys can make this latest release draft that would be great. We've same problem on CyberPanel rustic installation. 

@rehanone agree with you.

**Reply by simonsan** [reply]
@rehanone @usmannasir As we are working on the CI I marked the v0.5.4 as latest stable release (for automation) and the v0.6.0 as a pre-release version until we have implemented the workflow to automatically attach the artifacts.

Probably by the release of v0.6.1 we will have figured out the CI, so then we can mark that as the latest version. 👍🏽

**Reply by rehanone** [reply]
@simonsan Thank you!

**Reply by usmannasir** [reply]
@simonsan thanks a lot.
